### ISSUE TITLE
[Q]is it possible to choose the name of the sweep?

### ISSUE THREAD

**Issue by tommasodelorenzo** [wandb/client#2120]
Hello, I just discovered W&B few days ago and it is lovely.
I am running sweeps for my classification model using simpletransformers.
Is there a way one can choose the name of the sweep so that each run will be easily found in the dashboard?
What I am doing now is something like
```
sweep_config = {
    "name": "my_name",
    "method": "bayes",
    "metric": {"name": "accuracy", "goal": "maximize"},
    "parameters": {
        "num_train_epochs": {"distribution" : "int_uniform", "min": 4, "max" : 10},
        "learning_rate": {"distribution" : "log_normal", "mu": -6*log(10), "sigma": .5*log(10)},
    },
    "early_terminate": {"type": "hyperband", "min_iter": 4,},
}
sweep_id = wandb.sweep(sweep_config, project = "my_project_name")

def train():
    wandb.init()

    model = ClassificationModel(
        "bert",
        "dbmdz/bert-base-italian-uncased",
        num_labels=len(label_dict),
        args=model_args,
        sweep_config=wandb.config,
    )

    model.train_model(
        train_df=df_train,
        eval_df=df_val,
        accuracy=lambda truth, predictions: accuracy_score(
            truth, [round(p) for p in predictions]
        ),
    )
    
    model = nil
    wandb.join()

wandb.agent(sweep_id, train)
```
This way the sweep is correctly assigned to "my_project_name" project. However, the names of the sweep and single runs are chosen randomly. I'd like to have something like "my_name_1", "my_name_2", "my_name_3" as names of single runs in the sweep.
Is that possible? it seems that the `"name" : "my_name"` line in the `sweep_config` dictionary is not considered.

Thanks

**Comment by ariG23498** [comment]
Hey @tommasodelorenzo 

Below I have tried coding and describing the behaviour that we would notice if we use sweeps. Notice that I have taken a very minimal example to explain here.
```json
sweep_config = {
    "name": "my_name",
    "method": "bayes",
    "metric": {"name": "accuracy", "goal": "maximize"},
    "parameters": {
        "a":{
            "min":0.001,
            "max": 0.1,
        },
        "b":{
            "min": 32,
            "max": 64,
        },
    },
}
```
Here the `"name":"my_name"` key value pair defines the sweep's name that will run. In the screenshot below we see that the sweep name is indeed set to `my_name`.
![image](https://user-images.githubusercontent.com/36856589/116259238-4388c500-a793-11eb-8731-5344f2c3ebd0.png)

```python
sweep_id = wandb.sweep(sweep_config, entity="xxx", project="xxx")
```
This statement will initialise the sweep for the provided `entity` and `project`.

```python
def train():
    run = wandb.init()
    with run:
        run.log({
            "accuracy": run.config.a+run.config.b,
        })
wandb.agent(sweep_id, function=train)
```
This is the `train` method that the sweeper will repeatedly call. Here we see that the runs are named randomly, but if you click on the sweep you would be able to view only the runs of the particular sweep.
![image](https://user-images.githubusercontent.com/36856589/116261611-703ddc00-a795-11eb-9a05-f986d3f34806.png)


To demarcate the sweeps better in the general dashboard, I would recommend using the `group` option in the table view and select the `sweep` parameter there. You would be able to notice that all the runs are now grouped by their corresponding sweeps.
>A side note here, we do not group the sweeps with the name provided by the user, rather the unique id provided for the particular sweep.

![image](https://user-images.githubusercontent.com/36856589/116261291-1e955180-a795-11eb-95b9-7b8373a53824.png)

Does this seem to be a good workaround for the problem? Thanks for the ticket! 😄

**Comment by tommasodelorenzo** [comment]
Hello @ariG23498 

thanks for the answer. It does clarify the naming process that was a little obscure to me.
I do use the grouping-by-sweep option and there is where my question was eventually pointing to.
I'd like to have the sweeps grouped by the name I gave (if the same, than add a incremental number to it), so that I can easily recognized them when grouped.
I understand this is not possible as for now, and I should modify the source code to do so. Correct?

**Comment by ariG23498** [comment]
There is another workaround for the issue here:
We can use the `group` parameter in the `wandb.init` and pass the sweep name to the run. This will help us in using the grouping-by-group option which eventually will be pointing to the sweep name. This is indeed a redundant solution to the process but sure works.

I would ask for some time to confirm about the incremental naming of runs. If I come up with something, will certainly share here.

**Comment by ariG23498** [comment]
Hey @tommasodelorenzo 

The issue with incremental naming of runs is that multiple sweep agents could run the `train()` method. This way we would not be able to increment the run names properly. The workarounds I have provided are the best way to organise the runs. 

Closing this issue here. Thanks for the ticket 😄

**Comment by MightyStud** [comment]
> @tommasodelorenzo with regards to using custom names for the individual runs in a sweep, I found that having something like the following (pseudocode) in your `train` function works:
> 
> ```python
> def train():
>     with wandb.init() as run:
>         # Overwrite the random run names chosen by wandb
>         name_str = # Code for determining run names here
>         run.name = name_str
>         # Rest of code.
> ```
> 
> My particular interest was to have more informative run names such that `name_str` keys off of the hyperparameters found in `wandb.config` so that the learning rate, layer-size, etc are all in the run name. This doesn't address your desire for naming by sequential order, but it was close enough in topic that I thought you or others who come this this thread might find it helpful. I myself came across this thread when searching for how to accomplish the above.
> 
> ![sweep](https://user-images.githubusercontent.com/44747910/137157621-e31e03e9-5c08-4bfc-b97c-8ca8a66133e1.png)

can I pass an argument to the train function? I'd like to pass the name of the sweep from the agent, is that possible?

**Comment by idobz** [comment]
> @tommasodelorenzo with regards to using custom names for the individual runs in a sweep, I found that having something like the following (pseudocode) in your `train` function works:
> 
> ```python
> def train():
>     with wandb.init() as run:
>         # Overwrite the random run names chosen by wandb
>         name_str = # Code for determining run names here
>         run.name = name_str
>         # Rest of code.
> ```
> 
> My particular interest was to have more informative run names such that `name_str` keys off of the hyperparameters found in `wandb.config` so that the learning rate, layer-size, etc are all in the run name. This doesn't address your desire for naming by sequential order, but it was close enough in topic that I thought you or others who come this this thread might find it helpful. I myself came across this thread when searching for how to accomplish the above.
> 
> ![sweep](https://user-images.githubusercontent.com/44747910/137157621-e31e03e9-5c08-4bfc-b97c-8ca8a66133e1.png)

Building a little something on top of this. Assuming that you have some sort of config dictionary that manages the hyper parameters, and that in a sweep `wandb.config` over writes certain values per sweep, you can have make the `name_str` auto-add just the relevant HP values. Which is what, I think,  @tommasodelorenzo was after.


1. a `config_sweep.json` file that looks  like this: ```{
    "method": "grid",
    "parameters": {
        "dataset_numrows": {
            "values": [
                200000,
                400000
            ]
        }
    }
}```

2. `config_sweep.json`  is read by the wandb sweeping script, and also by the `train` function, where you specify:
```
        sweep_args = Namespace.from_filename("config_sweep.json")
        run = wandb.init()
        name_str = "_".join(
            [f"{key}_{wandb.config[key]}" for key in sweep_args.parameters.keys()]
        )
        run.name = run.name + "_" + name_str
```
